provider.go

```diff
@@ -74,15 +166,28 @@ func getAPI(hostAddr string, maxRetries int) (client.API, error) {
 func Provider() terraform.ResourceProvider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
-			"tunnel_address": &schema.Schema{
+			"driver": &schema.Schema{
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+				Default: "tunnel",
+				Description: "Driver to use to connect to Fleet. Can be tunnel or api.",
+			},
+			"endpoint": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Description: "Endpoint for Fleet.",
 			},
 			"connection_retries": &schema.Schema{
 				Type:     schema.TypeInt,
 				Optional: true,
 				Default:  12,
 			},
+			"etc_key_prefix": &schema.Schema{
```

Comment:
Should this be named `etcd_key_prefix` for consistency with `fleetctl`'s options?